pages/index.vue

```diff
@@ -10,31 +10,39 @@
           <b-button
             id="createReport"
             variant="success"
-            :disabled="isCreating"
+            :disabled="isBusy"
             @click="createReport"
           >
             Create report
-            <b-spinner v-show="isCreating" small></b-spinner>
+            <b-spinner v-show="isBusy" small></b-spinner>
           </b-button>
-          <b-button variant="info" :disabled="isCreating" @click="refreshTable">
+          <b-button variant="info" :disabled="isBusy" @click="refreshTable">
             Refresh
           </b-button>
+          <b-button
+            id="deleteReports"
+            variant="danger"
+            :disabled="isBusy"
+            @click="deleteReports"
+          >
+            Delete Reports
+          </b-button>
         </p>
+        <!-- TODO: better in a component of its own? -->
         <p>
           <b-alert :show="isError" dismissible variant="danger">
-            There was an error creating the report
+            {{ alertMessage }}
           </b-alert>
-          <b-alert :show="isCreated" dismissible variant="success">
-            Report successfully created
+          <b-alert :show="isSuccess" dismissible variant="success">
+            {{ alertMessage }}
           </b-alert>
-          <b-alert :show="isCreating" dismissible variant="warning">
-            Report creation takes about 30s to complete, please do not refresh
-            the page
+          <b-alert :show="isBusy" dismissible variant="warning">
+            {{ alertMessage }}
           </b-alert>
         </p>
         <b-table
           ref="reports_table"
-          :items="reportsProvider"
+          :items="items"
```

Comment:
We've spoke on slack so just for reference.. 

Looks like it was using a provider function before, which was quite nice. 
https://bootstrap-vue.org/docs/components/table#using-items-provider-functions, then section Example: Using callback to return data (asynchronous): (edited) 

 ` // Must return null or undefined to signal b-table that callback is being used
  return null`

Instead of the created function, if you return null from the reportsProvider I don't think you would have to recall this.provider() in refreshTable()